<launch>
<param name="use_sim_time" value="false"/>
<arg name="mp_name" default="moving_platform"/>
<arg name="publish_hz" default="100"/>
<arg name="x_offset" default="0"/>
<arg name="y_offset" default="0"/>
<arg name="z_offset" default="0"/>
<arg name="pwm_neutral" default="1500"/>
<arg name="max_amplitude" default="1"/>
  <group ns="$(arg mp_name)">
    <node name="ros_pid_fc_interface_node" pkg="moving_platform_control" type="ros_pid_fc_interface.py" output="screen" >
        <param name = "publish_hz" value = "$(arg publish_hz)" />
        <param name = "mp_name" value = "$(arg mp_name)" />
        <param name = "x_offset" value = "$(arg x_offset)" />
        <param name = "y_offset" value = "$(arg y_offset)" />
        <param name = "z_offset" value = "$(arg z_offset)" />
        <param name = "pwm_neutral" value = "$(arg pwm_neutral)" />
        <param name = "max_amplitude" value = "$(arg max_amplitude)" />

    </node> 
  </group>

  <node name="mp_controller" pkg="pid" type="controller" ns="mp_controller" output="screen" >
      <param name="node_name" value="pid_v_z" />
      <param name="Kp" value="2000" />
      <param name="Ki" value="900" />
      <param name="Kd" value="0" />
      <param name="upper_limit" value="500" />
      <param name="lower_limit" value="-500" />
      <param name="windup_limit" value="1" />
      <param name="max_loop_frequency" value="1000.0" />
      <param name="min_loop_frequency" value="1.0" />
      <param name="topic_from_controller" value="/vicon/moving_platform/controller/control_effort"/>
      <param name="topic_from_plant" value="/vicon/moving_platform/controller/state"/>
      <param name="setpoint_topic" value="/vicon/moving_platform/controller/setpoint"/>   
  </node>


</launch>